tsl/src/compression/create.c

```diff
@@ -118,15 +121,45 @@ build_columndefs(CompressionSettings *settings, Oid src_relid)
 	List *segmentby_column_defs = NIL;
 
 	Relation rel = table_open(src_relid, AccessShareLock);
+
+	/*
+	 * Check which columns have btree indexes. We will create sparse minmax
+	 * indexes for them in compressed chunk.
+	 */
+	Bitmapset *btree_columns = NULL;
+	ListCell *lc;
+	List *index_oids = RelationGetIndexList(rel);
+	foreach (lc, index_oids)
+	{
+		Oid index_oid = lfirst_oid(lc);
+		Relation index_rel = index_open(index_oid, AccessShareLock);
+		IndexInfo *index_info = BuildIndexInfo(index_rel);
+		index_close(index_rel, NoLock);
+
+		if (index_info->ii_Am != BTREE_AM_OID)
```

Comment:
Sorry for dumb question but what about partial indexes (aka `index_info->ii_Predicate != NIL`) ??

Reply:
Not sure about what's the best approach in this case, but at least even if we add this metadata, the result should still be correct. Maybe we should create minmax all the columns from predicate as well.

Reply:
Why are we tying this to btree?

Reply:
> Why are we tying this to btree?

The idea is that btree and sparse minmax index can be applied for similar types of queries.

If we have a gin index on a text column, the sparse minmax won't help the text search queries, so no point in creating it.

If we have a brin index, we have to look at the opclass -- if it's a bloom filter, minmax probably will be not so useful, but if it's a range brin, then minmax works. I haven't implemented this check yet.